var/spack/repos/builtin/packages/cntk/package.py

```diff
@@ -0,0 +1,113 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Cntk(Package):
+    """The Microsoft Cognitive Toolkit is a unified deep-learning toolkit
+    that describes neural networks as a series of computational steps
+    via a directed graph."""
+
+    homepage = "https://www.microsoft.com/en-us/research/product/cognitive-toolkit"
+    url      = "https://github.com/Microsoft/CNTK/archive/v2.0.beta15.0.tar.gz"
+
+    version('master', git='https://github.com/Microsoft/CNTK.git', branch='master')
+    version('2.0.rc1', url='https://github.com/Microsoft/CNTK/archive/v2.0.rc1.tar.gz')
+
+    variant('opencv', default=False, description="Enable OpenCV support.")
+    variant('kaldi', default=False, description="Enable Kaldi support.")
+    variant('asgd', default=True, description="Enable DataParallelASGD powered by Multiverso.")
+    variant('1bitsgd', default=False, description="Enable 1bitsgd support.")
+    variant('cuda', default=False, description="Enable CUDA support.")
+    variant('debug', default=False, description="Debug build.")
+
+    depends_on('libzip')
+    depends_on('openblas')
+    depends_on('mpi')
+    depends_on('boost')
+    depends_on('protobuf~shared')
+    # CNTK depends on kaldi@c02e8.
+    # See https://github.com/Microsoft/CNTK/blob/master/Tools/docker/CNTK-CPUOnly-Image/Dockerfile#L105-L125
+    depends_on('kaldi@c024e8', when='+kaldi')
+    depends_on('opencv', when='+opencv')
+    depends_on('cuda', when='+cuda')
+    depends_on('cub@1.4.1', when='+cuda')
+    depends_on('cudnn@5.1', when='+cuda')
+    depends_on('nccl', when='+cuda')
+    depends_on('cntk1bitsgd@c8b77d', when='+1bitsgd')
+    depends_on('multiverso@143187', when='+asgd')
```

Comment:
We don't normally restrict the version of a dependency unless we know that it only works with that version. Is that the case?